File: year2021/day01/python/solution.spec.py
from solution import part1, part2
import unittest
import logging

logging.basicConfig(level=logging.INFO)

class UnitTest(unittest.TestCase):

    @classmethod
    def setUp(self):
        logging.addLevelName(111, 'part1')
        logging.addLevelName(222, 'part2')
        
        with open('../example.txt') as file:
            self.input = [int(row) for row in file]


    def test_part1(self):
        print('\n----------------------------------------------------------------------')

        try:
            part_1, expected = part1(self.input), 7
            self.assertEqual(part_1, expected)
            logging.getLogger(' SUCCESS').log(111, '')
            
        except AssertionError:
            logging.getLogger(' FAILED').log(111, f' [result: {part_1}, expected: {expected}]')


    def test_part2(self):
        print('\n----------------------------------------------------------------------')
        
        try:
            part_2, expected = part2(self.input), 5
            self.assertEqual(part_2, expected)
            logging.getLogger(' SUCCESS').log(222, '')
            
        except AssertionError:
            logging.getLogger(' FAILED').log(222, f' [result: {part_2}, expected: {expected}]')


if __name__ == '__main__':
    unittest.main()

File: year2022/day06/java/SolutionSpec.java
import java.nio.file.*;
import java.util.*;

public class SolutionSpec {
  public static void main(String[] args) {
    try {
      Solution.setup(Files.readString(Paths.get("../example.txt")));
      test(1, Solution.part1(), 7);
      test(2, Solution.part2(), 19);
    } catch(Exception e) {
      e.printStackTrace();
    }
  }

  public static void test(int part, int solution, int expected) {
    assert solution == expected : "Part_" + part + ": FAILED - " + "[expected: " + expected + ", solution: " + solution + "]";
    System.out.println("Part_" + part + ": SUCCESS");
  }
}